File: 프로그래머스/LV2_기능개발.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    
    int pivot = 0;
    for (int i = 1; i <= 100; i++)
    {
        int num = 0;
        while(progresses[pivot] + (i * speeds[pivot]) >= 100 && progresses.size() > pivot)
        {
            pivot++;
            num++;
        }
        if (num)
        {
            answer.push_back(num);
        }
        if(progresses.size() == pivot)
            break;
    }
    return answer;
}

File: 자바스크립트/프로그래머스/LV2_기능개발.js
function solution(progresses, speeds) {
  var answer = [];

  while (progresses.length) {
    let finish = 0;
    while (progresses[0] >= 100) {
      progresses.shift();
      speeds.shift();
      finish += 1;
    }
    progresses.forEach((item, index) => (progresses[index] += speeds[index]));
    finish ? answer.push(finish) : null;
  }
  return answer;
}
